### ISSUE TITLE
Adding logical conditions to search (AND, OR, NOT etc)

### ISSUE THREAD

**Issue by Hapseleg** [PathOfBuildingCommunity/PathOfBuilding#160]
Hello

My idea is that you can use logical conditions.
For example, I want to find a unique item that gives life AND mana, right now I can only search on one of them.
For example if I wrote "life&&mana" (AND in js, c# and java) I would find both, or I could write "% increased maximum life || % increased maximum mana" then I would find items with one of them.

Im not sure if && and so on is the best solution, but its something that wont appear in item text and its easy for you to filter by

Keep up the great work

**Comment by kmoulins** [comment]
I suggest using a full blown regex at this point, with a choice of enabling/disabling it either via keyword (i.e. "r![regex]", maybe allowing to escape the !, so "r\![regex]" would search for "r![regex]" verbatim as is the case currently) or via a checkbox.

**Comment by Hapseleg** [comment]
Yea regex would be really nice, but add a checkbox for it imo, most people do not know the mystic arcane language of regex ;)

**Comment by Nostrademous** [comment]
@Wires77 wasn't this fixed a while ago?

**Comment by weregamer** [comment]
Oh, please please put regex back in. In the old PoB I could highlight all the nodes I wanted to target so I could find an optimal path quickly, but without the alternation construct it's a mess. For example, "(projectile)}(bow)|(resistance)" would help find paths to specific notables or keystones that I'm looking for with a Ranger, that hit the things I want to pick up on the way.

**Comment by Wires77** [comment]
> Oh, please please put regex back in. In the old PoB I could highlight all the nodes I wanted to target so I could find an optimal path quickly, but without the alternation construct it's a mess. For example, "(projectile)}(bow)|(resistance)" would help find paths to specific notables or keystones that I'm looking for with a Ranger, that hit the things I want to pick up on the way.

I think you're thinking of another program.  The original PoB never had regex searching, and in fact didn't even allow you to put parentheses in the search box.  For your specific scenario, you could type in `[pb][ro][ow]` to find most of the projectile/bow nodes.

I must've missed that this issue didn't get closed, but yes PoB does support pattern matching, which is similar to regex searching.  See here for a full rundown: https://www.lua.org/pil/20.2.html

**Comment by weregamer** [comment]
Very sad. This issue was not "have pattern matching", it was "support all
logical conditions including alternation". Lua matching specifically does
not support alternation. The documentation even points this out explicitly,
saying the calling code has to work around it.

Character groups are not the same thing, and your response proves my point
perfectly - read what I wanted, and compare it to what you produced.

I'm quite sure that the original PoB is what I'm thinking of. I discovered
that it was no longer being updated, and when I replaced it with "CoB
community" the regex checkbox went away.

**Comment by weregamer** [comment]
I'll have to try "show node power" to see what it can do; I haven't
explored that UI before.

Sorry for the possible heat in what I wrote before; I saw that first thing
in the morning after my alarm went off and before coffee ;) :(.

Strange that the Regex checkbox was not there in the main fork; I lived
with that option set the last time I got heavily into PoE which was all the
way back in Blight. Given that CoBC is in Lua, I can see that adding regex
to it would be nontrivial.

If node power doesn't float my boat, I'll write up a suggestion for
providing a dropdown OR list for searches, which could be done in the Lua
calling code along the lines suggested in their docs; the real code
complexity in regex is for patterns that require reparses, and I can't
imagine writing one of those for PoB (or really, anywhere - I may have
written one of those when grepping an entire SCC repository sometime in my
career, I suppose).
